spring-cloud-dataflow-configuration-metadata/src/main/java/org/springframework/cloud/dataflow/configuration/metadata/container/authorization/SecretToRegistryConfigurationConverter.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.configuration.metadata.container.authorization;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.cloud.dataflow.configuration.metadata.container.RegistryConfiguration;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.util.StringUtils;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * @author Christian Tzolov
+ */
+public class SecretToRegistryConfigurationConverter {
+
+	private static final Logger logger = LoggerFactory.getLogger(SecretToRegistryConfigurationConverter.class);
+
+	/**
+	 * {"auths":{"demo.goharbor.io":{"username":"admin","password":"Harbor12345","auth":"YWRtaW46SGFyYm9yMTIzNDU="}}}
+	 *
+	 * @param dockerconfigjson
+	 * @return
+	 */
+	public Map<String, RegistryConfiguration> convert(String dockerconfigjson) {
+
+		try {
+			Map authsMap = (Map) new ObjectMapper().readValue(dockerconfigjson, Map.class).get("auths");
+
+			Map<String, RegistryConfiguration> registryConfigurationMap = new HashMap<>();
+			for (Object registryUrl : authsMap.keySet()) {
+				RegistryConfiguration rc = new RegistryConfiguration();
+				rc.setRegistryHost(registryUrl.toString());
+				Map registryMap = (Map) authsMap.get(registryUrl.toString());
+				rc.setUser((String) registryMap.get("username"));
+				rc.setSecret((String) registryMap.get("password"));
+
+				String tokenAccessUrl = determineAuthorizationType(rc.getRegistryHost(), rc.getUser(), rc.getSecret());
+				if (StringUtils.isEmpty(tokenAccessUrl)) {
+					rc.setAuthorizationType(RegistryConfiguration.AuthorizationType.basicauth);
+				}
+				else {
+					rc.setAuthorizationType(RegistryConfiguration.AuthorizationType.dockerhub);
+					rc.getExtra().put("registryAuthUri", tokenAccessUrl);
+				}
+
+				logger.info("Registry Secret: " + rc.toString());
```

Comment:
might not be good to log this data

Reply:
I've changed the RegistryConfiguration's `toString()` implementation to prevent printing the secrets (e.g. password). 
IMO listing the provided and retrieved registry configurations is useful.  If necessary we can strip back the listed information even further, showing only the registry host uri and the authorization type?